File: src/main/java/chiloven/lukosbot2/commands/PingCommand.java
package chiloven.lukosbot2.commands;

import chiloven.lukosbot2.Constants;
import chiloven.lukosbot2.core.CommandSource;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;

import java.lang.management.ManagementFactory;
import java.lang.management.OperatingSystemMXBean;
import java.lang.management.RuntimeMXBean;
import java.time.Instant;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;

public class PingCommand implements BotCommand {
    private static final Constants CONST = new Constants();
    @Override
    public String name() {
        return "ping";
    }

    /**
     * Format uptime in seconds to d:hh:mm:ss
     *
     * @param seconds uptime in seconds
     * @return formatted uptime string
     */
    private static String formatUptime(long seconds) {
        long days = seconds / 86400;
        long hours = (seconds % 86400) / 3600;
        long minutes = (seconds % 3600) / 60;
        long secs = seconds % 60;
        return String.format("%d:%02d:%02d:%02d", days, hours, minutes, secs);
    }

    @Override
    public String usage() {
        return """
                用法：
                /ping # 返回 pong 及运行状态
                示例：
                /ping
                """;
    }

    @Override
    public String description() {
        return "健康检查：返回 pong 及运行状态";
    }

    @Override
    public void register(CommandDispatcher<CommandSource> dispatcher) {
        dispatcher.register(
                LiteralArgumentBuilder.<CommandSource>literal(name())
                        .executes(ctx -> {
                            ctx.getSource().reply(buildStatus());
                            return 1;
                        })
        );
    }

    private String buildStatus() {
        Runtime runtime = Runtime.getRuntime();
        long freeMem = runtime.freeMemory() / 1024 / 1024;
        long totalMem = runtime.totalMemory() / 1024 / 1024;
        long maxMem = runtime.maxMemory() / 1024 / 1024;

        RuntimeMXBean rtBean = ManagementFactory.getRuntimeMXBean();
        long uptimeSec = rtBean.getUptime() / 1000;
        String uptimeFmt = formatUptime(uptimeSec);

        OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
        String time = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
                .withZone(ZoneId.systemDefault())
                .format(Instant.now());

        return String.format("""
                        Pong 🏓
                        时间: %s
                        运行时间: %s
                        系统: %s %s
                        内存: 已用 %d MB / 总 %d MB (最大 %d MB)
                        线程数: %d
                        Java: %s | Spring: %s
                        TelegramBots: %s | JDA: %s | Shiro: %s
                        """,
                time,
                uptimeFmt,
                osBean.getName(), osBean.getVersion(),
                (totalMem - freeMem), totalMem, maxMem,
                Thread.activeCount(),
                CONST.javaVersion, CONST.springVersion,
                CONST.tgVersion, CONST.jdaVersion, CONST.shiroVersion
        );
    }
}

File: src/main/java/top/chiloven/lukosbot2/commands/impl/PingCommand.kt
package top.chiloven.lukosbot2.commands.impl

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service
import top.chiloven.lukosbot2.Constants
import top.chiloven.lukosbot2.commands.spec.bridge.SpecBotCommand
import top.chiloven.lukosbot2.commands.spec.dsl.botCommand
import top.chiloven.lukosbot2.util.StringUtils
import top.chiloven.lukosbot2.util.TimeUtils
import java.lang.management.ManagementFactory
import java.time.Instant
import java.time.ZoneId

@Service
@ConditionalOnProperty(
    prefix = "lukos.commands.control",
    name = ["ping"],
    havingValue = "true",
    matchIfMissing = true
)
class PingCommand : SpecBotCommand() {

    override fun spec() = botCommand("ping") {
        description = "返回机器人的运行状态与版本信息"

        execute {
            reply(buildStatus())
        }

        syntax("检测机器人在线状态")
        example("ping")
    }

    private fun buildStatus(): String {
        val runtime = Runtime.getRuntime()

        val totalMem = StringUtils.fmtBytes(runtime.totalMemory(), 2)
        val maxMem = StringUtils.fmtBytes(runtime.maxMemory(), 2)
        val usedMem = StringUtils.fmtBytes(runtime.totalMemory() - runtime.freeMemory(), 2)

        val rtBean = ManagementFactory.getRuntimeMXBean()
        val uptimeFmt = TimeUtils.formatUptime(rtBean.uptime / 1000)

        val osBean = ManagementFactory.getOperatingSystemMXBean()
        val time = TimeUtils.dtf().withZone(ZoneId.systemDefault()).format(Instant.now())

        return """
            Pong！$time
            ${Constants.APP_NAME} ${Constants.VERSION}
            运行时间：$uptimeFmt
            系统：${osBean.name} ${osBean.version}
            内存：$usedMem / $totalMem（最大 $maxMem）
            Java：${Constants.javaVersion} | Kotlin：${Constants.kotlinVersion} | Spring Boot：${Constants.springBootVersion}
            TelegramBots: ${Constants.tgVersion} | JDA: ${Constants.jdaVersion} | Shiro: ${Constants.shiroVersion}
        """.trimIndent()
    }

}
